addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-service.js

```diff
@@ -226,13 +228,25 @@ class EnvironmentScService extends Service {
     };
     const sagemakerRealtimeStatus = await this.pollSageMakerRealtimeStatus(roleArn, externalId);
     const sagemakerUpdated = {};
-    _.forEach(sagemakerInstances, async (existingEnvRecord, key) => {
+    const sagemakerKeys = Object.keys(sagemakerInstances);
+    for (let i = 0; i < sagemakerKeys.length; i += 1) {
+      const key = sagemakerKeys[i];
+      const existingEnvRecord = sagemakerInstances[key];
       const expectedDDBStatus = SageMakerStatusMap[sagemakerRealtimeStatus[key]];
+      // eslint-disable-next-line no-await-in-loop
       const updateStatusResult = await this.updateStatus(requestContext, existingEnvRecord, expectedDDBStatus);
       if (updateStatusResult) {
         sagemakerUpdated[key] = updateStatusResult;
       }
-    });
+    }
+
+    // _.forEach(sagemakerInstances, async (existingEnvRecord, key) => {
```

Comment:
I think this for loop might not behave as we think. It might not `await` for all the update requests to complete, or it might be executing them in parallel.

For this part, please double check if the `for (let i = 0.....` code is needed or if we can still use this `forEach` loop.

For more [info](https://stackoverflow.com/a/37576787)